src/openrct2/entity/Peep.cpp

```diff
@@ -1608,7 +1608,11 @@ void Peep::FormatNameTo(Formatter& ft) const
 
         if ((!isStaff && showGuestNames) || (isStaff && showStaffNames))
         {
-            auto realNameStringId = GetRealNameStringIDFromPeepID(PeepId);
+            auto nameId = PeepId;
+            if (isStaff)
+                nameId *= 256 * EnumValue(staff->AssignedStaffType) + 1;
+
+            auto realNameStringId = GetRealNameStringIDFromPeepID(nameId);
```

Comment:
Is the ternary much here? :D
```suggestion
            auto nameId = isStaff ? (PeepId * 256 * EnumValue(staff->AssignedStaffType) + 1) : PeepId;
            auto realNameStringId = GetRealNameStringIDFromPeepID(nameId);
```

In any case, I think a comment stating what this offset is doing will be good for future readers.

Reply:
Haha, I considered a ternary, but I don't think it improves the readability tbh.

Agreed a comment would be good. I'll add one tomorrow.

Reply:
@tupaschoal Does this sound good to you?
```suggestion
            auto nameId = PeepId;
            if (isStaff)
            {
                // Prevent staff from getting the same names by offsetting the name table based on staff type.
                nameId *= 256 * EnumValue(staff->AssignedStaffType) + 1;
            }

            auto realNameStringId = GetRealNameStringIDFromPeepID(nameId);
```

Reply:
It does!